gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTooFarFromTripShapeValidatorTest.java

```diff
@@ -0,0 +1,164 @@
+        return GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+    }
+
+    private static GtfsShapeTableContainer createShapeTable(String[] shapeIds, double[] shapePtLats, double[] shapePtLons, int[] shapePtSequences, NoticeContainer noticeContainer) {
+        Preconditions.checkArgument(shapeIds.length == shapePtLats.length, "shapeIds.length must be equal to shapePtLats.length");
+        Preconditions.checkArgument(shapePtLats.length == shapePtLons.length, "shapePtLats.length must be equal to shapePtLons.length");
+        Preconditions.checkArgument(shapeIds.length == shapePtSequences.length, "shapeIds.length must be equal to shapePtSequences.length");
+        ArrayList<GtfsShape> shapes = new ArrayList<>();
+        shapes.ensureCapacity(shapePtLats.length);
+        for (int i = 0; i < shapePtLats.length; i++) {
+            shapes.add(createShape(shapes.size() + 1, shapeIds[i], shapePtLats[i], shapePtLons[i], shapePtSequences[i]));
+        }
+        return GtfsShapeTableContainer.forEntities(shapes, noticeContainer);
+    }
+
+    private static GtfsTripTableContainer createTripTable(String[] tripIds, String[] shapeIds, NoticeContainer noticeContainer) {
+        Preconditions.checkArgument(tripIds.length == shapeIds.length, "tripIds.length must be equal to shapeIds.length");
+        ArrayList<GtfsTrip> trips = new ArrayList<>();
+        trips.ensureCapacity(tripIds.length);
+        for (int i = 0; i < tripIds.length; i++) {
+            trips.add(createTrip(trips.size() + 1, tripIds[i], shapeIds[i]));
+        }
+        return GtfsTripTableContainer.forEntities(trips, noticeContainer);
+    }
+
+    @Test
+    public void stopWithinTripShapeShouldNotGenerateNotice() {
+        // See map of trip shape and stops (in GeoJSON) at https://gist.github.com/barbeau/d9c0b90a26a3e2ba105cae5f0e8aec4a
+        final NoticeContainer noticeContainer = new NoticeContainer();
+        StopTooFarFromTripShapeValidator validator = new StopTooFarFromTripShapeValidator();
+        validator.stopTable = createStopTable(
+            new String[]{"1001", "1002"},
+            new double[]{28.05811731042478d, 28.05812364854794d},
+            new double[]{-82.41616877502503d, -82.41617370439423d},
+            new GtfsLocationType[]{GtfsLocationType.STOP, GtfsLocationType.STOP},
+            noticeContainer);
+        validator.stopTimeTable = createStopTimeTable(
+            new String[]{"trip1", "trip1"},
+            new String[]{"1001", "1002"},
+            new int[]{1, 2},
+            noticeContainer);
+        validator.shapeTable = createShapeTable(
+            new String[]{"shape1", "shape1", "shape1", "shape1", "shape1"},
+            new double[]{28.05724310653972d, 28.05746701492806d, 28.05800068503469d, 28.05808869825447d, 28.05809979887893d},
+            new double[]{-82.41350776611507d, -82.41493135129478d, -82.4159394137605d, -82.41648754043338d, -82.41773971025437d},
+            new int[]{1, 2, 3, 4, 5},
```

Comment:
You could make the code more readable if you used something similar to https://www.baeldung.com/introduction-to-autovalue

Reply:
I have read through the introduction for AutoValue and got some ideas. Do you mean you want me to insert the data entity by entity (e.g. one line for a GtfsStopTime, one line for a GtfsShape), rather than putting all data for one field for all entities of a Gtfs type together?

Reply:
Currently, the arrays of data for different fields are as input to createStopTable. Sorry, but I'm still confused what I need to adjust here or for the "createStopTable" helper method? What is suggested to be put as input?

Reply:
As there is already a builder for the relevant elements, the code would be more readable if the test explicitly creates the elements (using the builders), instead of indirectly using a function zipping different attributes

Reply:
Yes. I have now changed to directly use builders to create GTFS elements, rather than having extra helper methods to zip values of one field together.